packages/run-protocol/test/amm/vpool-xyk-amm/setup.js

```diff
@@ -13,35 +13,49 @@ import {
 import * as Collect from '../../../src/collect.js';
 import {
   setupAmm,
+  setupReserve,
   startEconomicCommittee,
 } from '../../../src/proposals/econ-behaviors.js';
 import { installGovernance, provideBundle } from '../../supports.js';
+import { makeTracer } from '../../../src/makeTracer.js';
 
 const ammRoot = './src/vpool-xyk-amm/multipoolMarketMaker.js'; // package relative
+const reserveRoot = './src/reserve/assetReserve.js'; // package relative
 
-export const setUpZoeForTest = async () => {
+const trace = makeTracer('AmmTS', false);
+
+export const setUpZoeForTest = () => {
   const { makeFar } = makeLoopback('zoeTest');
 
   const { zoeService, feeMintAccess: nonFarFeeMintAccess } = makeZoeKit(
     makeFakeVatAdmin(() => {}).admin,
   );
   /** @type {ERef<ZoeService>} */
   const zoe = makeFar(zoeService);
-  const feeMintAccess = await makeFar(nonFarFeeMintAccess);
+  const feeMintAccess = makeFar(nonFarFeeMintAccess);
   return {
     zoe,
     feeMintAccess,
   };
 };
 harden(setUpZoeForTest);
+/**
+ * @typedef {ReturnType<typeof setUpZoeForTest>} FarZoeKit
+ */
 
+/**
+ *
+ * @param {TimerService} timer
+ * @param {FarZoeKit} [farZoeKit]
+ */
 export const setupAMMBootstrap = async (
   timer = buildManualTimer(console.log),
-  zoe,
+  farZoeKit,
 ) => {
-  if (!zoe) {
-    ({ zoe } = await setUpZoeForTest());
+  if (!farZoeKit) {
+    farZoeKit = await setUpZoeForTest();
```

Comment:
Can this be simplified back out, too? [I thought you said similar code was added elsewhere to assist in debugging, but was no longer useful.]

Reply:
I don't think so. In this case the need for `farZoeKit` had to be wired through.